compiler/rustc_const_eval/src/check_consts/ops.rs

```diff
@@ -355,20 +357,82 @@ fn build_error_for_const_call<'tcx>(
                 non_or_conditionally,
             })
         }
-        _ => ccx.dcx().create_err(errors::NonConstFnCall {
-            span,
-            def_descr: ccx.tcx.def_descr(callee),
-            def_path_str: ccx.tcx.def_path_str_with_args(callee, args),
-            kind: ccx.const_kind(),
-            non_or_conditionally,
-        }),
+        _ => {
+            let def_descr = ccx.tcx.def_descr(callee);
+            let mut err = ccx.dcx().create_err(errors::NonConstFnCall {
+                span,
+                def_descr,
+                def_path_str: ccx.tcx.def_path_str_with_args(callee, args),
+                kind: ccx.const_kind(),
+                non_or_conditionally,
+            });
+            let context_span = ccx.tcx.def_span(ccx.def_id());
+            err.span_label(context_span, format!(
```

Comment:
Why did you change this to a label only for this case, and keep it as a note for the rest? Having special logic changing it between a note and a label seems too special cased. I'd prefer if you reverted it.

Reply:
This line is pretty long for a label too.

Reply:
> I wanted to point at the constant context in the output in some way, as it isn't always trivially obvious why a constant function was expected.

I don't think the users need this to be a label. There's a tradeoff here with verbosity and I think this both complicates the implementation by having that `note` boolean and also splits the presentation of the diagnostic.